util/src/main/scala/scala/scalanative/util/Scope.scala

```diff
@@ -39,21 +41,34 @@ object Scope {
       throw new UnsupportedOperationException("Can't close forever Scope.")
   }
 
+  /** Unsafe manually managed scope.*/
+  def unsafe: Scope = new Impl {}
+
   private sealed class Impl extends Scope {
-    private[this] var resources: List[Resource] = Nil
+    type Resources = List[Resource]
+
+    private[this] val resources = new AtomicReference[Resources](Nil)
+
+    private def set(fn: Resources => Resources): Resources =
```

Comment:
With the suggested implementation of `close()`, this will only be used in `acquire`. I suggest to inline it there, as it will be clearer that it's correct in the context of `acquire` than in a generic context where we don't know what `fn` does.